libraries/Adapters/Microsoft.Bot.Builder.Adapters.Twilio/TwilioHelper.cs

```diff
@@ -30,19 +30,19 @@ internal static class TwilioHelper
         /// Creates Twilio SMS message options object from a Bot Framework <see cref="Activity"/>.
         /// </summary>
         /// <param name="activity">The activity.</param>
-        /// <param name="twilioNumber">The Twilio phone number assigned to the bot.</param>
+        /// <param name="twilioNumber">Optional. The Twilio phone number assigned to the bot. If not provided, defaults to Activity.From.Id to allow WhatsApp and other future integrations.</param>
         /// <returns>The Twilio message options object.</returns>
         /// <seealso cref="TwilioAdapter.SendActivitiesAsync(ITurnContext, Activity[], System.Threading.CancellationToken)"/>
-        public static TwilioMessageOptions ActivityToTwilio(Activity activity, string twilioNumber)
+        public static TwilioMessageOptions ActivityToTwilio(Activity activity, string twilioNumber = null)
         {
             if (activity == null)
             {
                 throw new ArgumentNullException(nameof(activity));
             }
 
-            if (string.IsNullOrWhiteSpace(twilioNumber))
+            if (string.IsNullOrWhiteSpace(twilioNumber) && string.IsNullOrWhiteSpace(activity.From?.Id))
             {
-                throw new ArgumentNullException(nameof(twilioNumber));
+                throw new ArgumentException($"Either {nameof(twilioNumber)} or {nameof(activity.From.Id)} must be provided.");
```

Comment:
>Either {nameof(twilioNumber)} or {nameof(activity.From.Id)} must be provided. [](http://example.com/codeflow?start=46&length=77)

Looks like the client could pass twilioNumber and get here. If that happened, the message could be confusing.

Reply:
I'm not sure what you mean. To hit this Exception, `twilioNumber` AND `activity.From.Id` would have to be `nullOrWhitespace`.

Reply:
Because we are not passing twilioNumber, it will always appear as null here.

Reply:
Ah, I see what you mean. Yes, as our code uses it, it will always be `null`. However, `activity.From.Id` *should* also always be populated from our use, so this shouldn't ever be hit by our own code. This was more just to cover the case where a developer might call this more directly. I'm open to other ideas.